packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -138,6 +138,11 @@ sealed class _DebugSemanticsRoleChecks {
     SemanticsRole.status => _noLiveRegion,
     SemanticsRole.list => _noCheckRequired,
     SemanticsRole.listItem => _semanticsListItem,
+    SemanticsRole.complementary => _noCheckRequired,
```

Comment:
According to https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/

The requirements are
*complementary landmarks should be top level landmarks (e.g. not contained within any other landmarks).
*If the complementary content is not related to the main content, a more general role should be assigned (e.g. region).
*If a page includes more than one complementary landmark, each should have a unique label (see [Step 3](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/#aria_lh_step3) above).

Reply:
all other have similar traits, we should have appropriate check here as well.

Reply:
Ah I didn't know this website before! Thanks for letting me know. Will update!

Reply:
Based on the requirements from the website, complementary, contentInfo and main should not be contained within any other landmark roles. So I added a check for them.

I don't see a "must have" requirement for `navigation`([here](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/#:~:text=Main%20Landmark%20Example-,Navigation,-Navigation%20landmarks%20provide)), so I leave the `_noCheckRequired` for the role as is.